### PR TITLE
SI-7331 importers now import tree positions

### PR DESCRIPTION
**Pull request by xeno-by**
This fixes the problem with toolbox-parsed trees not having positions.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @retronym

**Comment by retronym** [comment]
LGTM

**Comment by adriaanm** [comment]
PLS REBUILD/pr-checkin-per-commit@102cd3e

File: src/reflect/scala/reflect/internal/Importers.scala [hunk 1]
```diff
@@ -443,6 +443,7 @@ trait Importers extends api.Importers { self: SymbolTable =>
         }
       })
       tryFixup()
+      mytree.pos = importPosition(tree.pos)
       mytree
```
Comment (paulp) [inline src/reflect/scala/reflect/internal/Importers.scala]:
```
mytree setPos importPosition(tree.pos)
```

I would like to deprecate pos_= ; please use setPos.

Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Importers.scala]: done

File: src/reflect/scala/reflect/internal/Importers.scala [hunk 2]
```diff
@@ -443,6 +443,8 @@ trait Importers extends api.Importers { self: SymbolTable =>
         }
       })
       tryFixup()
+      if (mytree ne emptyValDef)
+        mytree setPos importPosition(tree.pos)
       mytree
```
Comment (paulp) [inline src/reflect/scala/reflect/internal/Importers.scala:448]: Why on earth does CannotHaveAttrs throw an exception on setPos but not pos_=, while throwing an exception on tpe_= but not setType? Why does tpe_= allow for NoType, but setPos doesn''t allow for NoPosition? I'd have a better chance at predicting keno numbers than I would at working out this logic.

Is there some occasion where you will be glad to have called setPos on emptyValDef and brought about an exception? If there isn't, why does every call site have to guard against emptyValDef?

**Comment by adriaanm** [comment]
Ping. I'd like to see a reply to the [comment](https://github.com/scala/scala/pull/2357#discussion_r3685600) before merging.

**Comment by xeno-by** [comment]
Pong. Will take a look today evening.

File: src/reflect/scala/reflect/internal/Importers.scala (hunk 2 above)
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Importers.scala:448]: These are all very valid questions, so I'm going have to dig deeper. Unfortunately I'm currenty full-time writing a paper, so I'll close this one and reopen when I have more time.

**Pull request closed without merging**